src/main/integrations/preload-injection.ts

```diff
@@ -4,12 +4,34 @@ import { Integration } from '@sentry/types';
 import { logger } from '@sentry/utils';
 import { app } from 'electron';
 import { existsSync } from 'fs';
-import { isAbsolute } from 'path';
+import { isAbsolute, resolve } from 'path';
+import { fileURLToPath } from 'url';
 
 import { IPCMode } from '../../common';
 import { rendererRequiresCrashReporterStart } from '../electron-normalize';
 import { ElectronMainOptionsInternal } from '../sdk';
 
+// After bundling with webpack, require.resolve can return number so we include that in the types
+// to ensure we check for that!
+function getPreloadPath(): string | number | undefined {
+  try {
+    return rendererRequiresCrashReporterStart()
+      ? require.resolve('../../preload/legacy.js')
+      : require.resolve('../../preload/index.js');
+  } catch (_) {
+    try {
+      // This could be ESM
+      const currentDir = fileURLToPath(import.meta.url);
+      // Use the CJS preload
+      return resolve(currentDir, '..', '..', '..', '..', 'preload', 'index.js');
```

Comment:
I wonder if this will break with any bundlers, but if the e2e tests pass we should be fine I guess.